src/lib/ubisys.ts

```diff
@@ -1,12 +1,151 @@
+                ubisysIntegralFactor: {ID: 0x0022, type: Zcl.DataType.int16,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+            },
+            commands: {},
+            commandsResponse: {},
+        },
+    ),
+    addClusterGenLevelCtrl: () => deviceAddCluster(
+        'genLevelCtrl',
+        {
+            ID: 0x0008,
+            attributes: {
+                // D1(-R)
+                ubisysMinimumOnLevel: {ID: 0x0000, type: Zcl.DataType.uint8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                // H10
+                ubisysValveType: {ID: 0x0001, type: Zcl.DataType.bitmap8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                ubisysCyclePeriod: {ID: 0x0002, type: Zcl.DataType.uint8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                ubisysSeason: {ID: 0x0003, type: Zcl.DataType.enum8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                ubisysBackupLevel: {ID: 0x0004, type: Zcl.DataType.uint8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                ubisysAlternateBackupLevel: {ID: 0x0005, type: Zcl.DataType.uint8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                ubisysLowerRange: {ID: 0x0006, type: Zcl.DataType.uint8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                ubisysUpperRange: {ID: 0x0007, type: Zcl.DataType.uint8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                ubisysPumpThresholdOn: {ID: 0x0008, type: Zcl.DataType.uint8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                ubisysPumpThresholdOff: {ID: 0x0009, type: Zcl.DataType.uint8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                ubisysHeatingDemandEnableThreshold: {ID: 0x000A, type: Zcl.DataType.uint8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                ubisysHeatingDemandDisableThreshold: {ID: 0x000B, type: Zcl.DataType.uint8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                ubisysCoolingDemandEnableThreshold: {ID: 0x000C, type: Zcl.DataType.uint8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+                ubisysCoolingDemandDisableThreshold: {ID: 0x000D, type: Zcl.DataType.uint8,
+                    manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH},
+            },
+            commands: {},
+            commandsResponse: {},
+        },
+    ),
+    addClusterClosuresWindowCovering: () => deviceAddCluster(
```

Comment:
@Koenkk nog een vraagje ivm de merge order...

Hoe loopt dit als ik hier top level `manufacturerCode: Zcl.ManufacturerCode.UBISYS_TECHNOLOGIES_GMBH,` bij zet? Dan kan het in theory weg bij de attributen.

Maar het was me niet duidelijk wat er dan gebeurd met de standaard ZCL attributen in closuresWindowCovering.

Reply:
The manufacturer code from the custom cluster will always be used ([code](https://github.com/Koenkk/zigbee-herdsman/blob/86c080ec8e522db7e774025d8592e4f27272c1de/src/controller/model/device.ts#L823C17-L823C60)). So this PR is OK.